zone/spells.cpp

```diff
@@ -2343,6 +2344,11 @@ bool Mob::SpellFinished(uint16 spell_id, Mob *spell_target, CastingSlot slot, ui
 	}
 
 	// check line of sight to target if it's a detrimental spell
+	if (!CheckWaterLoS(this, spell_target)) {
```

Comment:
The if block below this checks quite a few things like making sure spell_target is valid and other things. I would combine this check into the below if by grouping the regular LOS check and the Water LOS check with an OR inside of parenthesis.
`if(!spells[spell_id].npc_no_los && spell_target && IsDetrimentalSpell(spell_id) && (!CheckLosFN(spell_target) || !CheckWaterLoS(this, spell_target)) && !IsHarmonySpell(spell_id) && spells[spell_id].target_type != ST_TargetOptional)`